tests/templates/kuttl/smoke/31-assert.yaml

```
@@ -0,0 +1,11 @@
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+timeout: 600
+commands:
+  #
+  # Test envOverrides
+  #
+  - script: |
+      kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data."superset_config.py"' | grep "COMMON_HEADER_VAR = "group-value""
+      kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data."superset_config.py"' | grep "ROLE_FOOTER_VAR = "role-value""
```

Can we please also assert that `ROLE_HEADER_VAR` is not included?

---

This will be overwritten in the process.

Means first we have
```
      EXPERIMENTAL_FILE_HEADER: |
          COMMON_HEADER_VAR = "role-value"
          ROLE_HEADER_VAR = "role-value"
```
After role group:
```
        EXPERIMENTAL_FILE_HEADER: |
          COMMON_HEADER_VAR = "group-value"
```

We could assert if it's not present anymore. Imo it's sufficient then to test if the value of `COMMON_HEADER_VAR` is overwritten.

---

done https://github.com/stackabletech/superset-operator/pull/533/commits/0d25aeb87b9396f0484d606266cb2cb2a3d5c335

---

There are unescaped double-quotes inside double-quotes. I suggest using single quotes when variable expansion is not needed:

```suggestion
      kubectl -n $NAMESPACE get cm superset-node-default -o yaml | yq -e '.data["superset_config.py"]' | grep 'ROLE_FOOTER_VAR = "role-value"'
```

Also applies to other lines

---

I will post a separate comment with what I suggest the whole script should look like

---

+1 on using `grep -v` , was thinking the same :+1: